File: library-management/python/tests/member_builder.py
from __future__ import annotations

from itertools import count

from library_management import Member

_next_id = count(1000)


class MemberBuilder:
    def __init__(self) -> None:
        self._id = next(_next_id)
        self._name = "Alex Reader"

    def named(self, name: str) -> "MemberBuilder":
        self._name = name
        return self

    def with_id(self, id: int) -> "MemberBuilder":
        self._id = id
        return self

    def build(self) -> Member:
        return Member(self._id, self._name)

File: library-management/typescript/tests/MemberBuilder.ts
import { Member } from '../src/Member.js';

let nextId = 1000;

export class MemberBuilder {
  private _id = ++nextId;
  private _name = 'Alex Reader';

  named(name: string): this { this._name = name; return this; }
  withId(id: number): this { this._id = id; return this; }

  build(): Member { return new Member(this._id, this._name); }
}
